pubsublite/producer.go

```diff
@@ -117,24 +99,17 @@ func NewProducer(cfg ProducerConfig) (*Producer, error) {
 	if err := cfg.Validate(); err != nil {
 		return nil, fmt.Errorf("pubsublite: invalid producer config: %w", err)
 	}
-
-	tracerProvider := cfg.TracerProvider
-	if tracerProvider == nil {
-		tracerProvider = otel.GetTracerProvider()
-	}
-
 	p := &Producer{
 		cfg:    cfg,
 		closed: make(chan struct{}),
 		// NOTE(marclop) should the channel size be dynamic? 1000 is an arbitrary
 		// number, but it must be greater than 0, so async produces don't block.
 		responses: make(chan []resTopic, 1000),
-		tracer:    tracerProvider.Tracer("pubsublite"),
+		tracer:    cfg.tracerProvider().Tracer("pubsublite"),
 
 		project: cfg.Project,
 		region:  cfg.Region,
```

Comment:
We don't seem to be using these 2 fields anymore.

Reply:
Indeed, thanks. Probably should set up staticcheck at some point.